tests/test_axis.py

```diff
@@ -543,6 +538,70 @@ def fulfill(internal_format="not AxisNow"):
         fulfillment = fulfill(internal_format="irrelevant")
         assert isinstance(fulfillment, Axis360FulfillmentInfo)
 
+    def test_fulfill_without_subsequent_availability_call(self):
+        """
+        GIVEN: A patron requesting loan fulfillmlent
+        WHEN:  Fulfilling the loan
+        THEN:  The transaction_id is saved to loan.external_identifier
+        """
+        _, pool = self._edition(
+            identifier_type=Identifier.AXIS_360_ID,
+            identifier_id='0015176429',
+            data_source_name=DataSource.AXIS_360,
+            with_license_pool=True
+        )
+
+        patron = self._patron()
+        patron.authorization_identifier = "a barcode"
+        # Ensure a Loan exists for this patron for future calls
+        create(self._db, Loan, patron=patron, license_pool=pool) 
+        data = self.sample_data("availability_with_axisnow_fulfillment.xml")
+        data = data.replace(b"0016820953", pool.identifier.identifier.encode("utf8"))
+        self.api.queue_response(200, content=data)
+        first_fulfillment = self.api.fulfill(patron, "pin", pool, "AxisNow")
```

Comment:
Verify that the external_identifier has been set after the first fulfillment, rather than waiting for the end of the test.